src/test/java/app/daos/impl/GameDAOImplTest.java

```diff
@@ -0,0 +1,121 @@
+package app.daos.impl;
+
+import app.Populator;
+import app.config.HibernateConfig;
+import app.entities.Game;
+import app.entities.Genre;
+import app.entities.Platform;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityNotFoundException;
+import org.junit.jupiter.api.*;
+
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.hamcrest.MatcherAssert.*;
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+class GameDAOImplTest {
+
+    private static Populator populator;
+    private static GameDAOImpl gameDAO;
+    private List<Genre> genres;
+    private List<Platform> platforms;
+    private List<Game> games;
+
+    @BeforeAll
+    static void beforeAll() {
+        EntityManagerFactory emf = HibernateConfig.getEntityManagerFactoryForTest();
+
+        populator = new Populator(emf);
+        gameDAO = GameDAOImpl.getInstance(emf);
+    }
+
+    @BeforeEach
+    void setUp() {
+        genres = populator.createGenres();
+        populator.persist(genres);
+
+        platforms = populator.createPlatforms();
+        populator.persist(platforms);
+
+        games = populator.createGames(genres, platforms);
+        populator.persist(games);
+    }
+
+    @AfterEach
+    void tearDown() {
+        populator.cleanup(Game.class);
+        populator.cleanup(Genre.class);
+        populator.cleanup(Platform.class);
+    }
+
+    @Test
+    void create() {
+        List<Game> sortedGames = games.stream()
+                .sorted(Comparator.comparing(Game::getId))
+                .toList();
+
+        Long lastId = sortedGames.get(sortedGames.size() - 1).getId();
```

Comment:
Replace with a more direct method of obtaining the last game: `Game lastGame = games.stream().max(Comparator.comparing(Game::getId)).orElseThrow();`